File: backend/src/Sgr/V1/ReciboBundle/Entity/Pessoa.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

namespace App\Sgr\V1\ReciboBundle\Entity;

/**
 * Description of UsuarioGrupo
 *
 * @author odenilton
 */
use Doctrine\ORM\Mapping as ORM;
use App\Sgr\BaseBundle\Utils\Configurator;

/**
 * @ORM\Entity
 * @ORM\Table(name="pessoa")
 * @ORM\Entity(repositoryClass="App\Sgr\V1\ReciboBundle\Repository\PessoaRepository")
 */
class Pessoa {

    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue
     * @var int
     */
    private $id;

    /**
     * @ORM\Column(name="nome", type="string", nullable=true)
     * @var string
     */
    private $nome;

    /**
     * @ORM\Column(name="nit", type="string", nullable=true)
     * @var string
     */
    private $nit;

    /**
     * @ORM\Column(name="cpf", type="string", nullable=true)
     * @var string
     */
    private $cpf;

    /**
     * @ORM\Column(name="rg", type="string", nullable=true)
     * @var string
     */
    private $rg;

    /**
     * @ORM\Column(name="rgOrgaoExpeditor", type="string", nullable=true)
     * @var string
     */
    private $rgOrgaoExpeditor;

    /**
     * @ORM\Column(name="rgUfOrgaoExpeditor", type="integer", nullable=true)
     * @var string
     */
    private $rgUfOrgaoExpeditor;

    /**
     * @ORM\Column(name="rgDataExpedicao", type="string", nullable=true)
     * @var string
     */
    private $rgDataExpedicao;

    /**
     * @ORM\Column(name="nomeMae", type="string", nullable=true)
     * @var string
     */
    private $nomeMae;

    /**
     * @ORM\Column(name="nomePai", type="string", nullable=true)
     * @var string
     */
    private $nomePai;

    /**
     * @ORM\Column(name="dataNascimento", type="string", nullable=true)
     * @var string
     */
    private $dataNascimento;

    /**
     * @ORM\Column(name="sexo", type="integer", nullable=true)
     * @var string
     */
    private $sexo;

    /**
     * @ORM\Column(name="naturalidade", type="string", nullable=true)
     * @var string
     */
    private $naturalidade;

    /**
     * @ORM\Column(name="ufNaturalidade", type="integer", nullable=true)
     * @var string
     */
    private $ufNaturalidade;

    /**
     * @ORM\Column(name="telefone", type="string", nullable=true)
     * @var string
     */
    private $telefone;

    /**
     * @ORM\Column(name="formaPagamento", type="integer", nullable=true)
     * @var string
     */
    private $formaPagamento;

    /**
     * @ORM\Column(name="banco", type="integer", nullable=true)
     * @var string
     */
    private $banco;

    /**
     * @ORM\Column(name="agencia", type="string", nullable=true)
     * @var string
     */
    private $agencia;

    /**
     * @ORM\Column(name="operacao", type="string", nullable=true)
     * @var string
     */
    private $operacao;

    /**
     * @ORM\Column(name="digitoAgencia", type="string", nullable=true)
     * @var string
     */
    private $digitoAgencia;

    /**
     * @ORM\Column(name="conta", type="string", nullable=true)
     * @var string
     */
    private $conta;

    /**
     * @ORM\Column(name="digitoConta", type="string", nullable=true)
     * @var string
     */
    private $digitoConta;

    /**
     * @ORM\Column(name="tipoConta", type="integer", nullable=true)
     * @var string
     */
    private $tipoConta;

    /**
     * @ORM\OneToOne(targetEntity="App\Sgr\V1\ReciboBundle\Entity\PessoaEndereco", cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="endereco_id", referencedColumnName="id")
     */
    protected $endereco;

    public function __construct($options = null) {
        Configurator::configure($this, $options);
    }

    function getId() {
        return $this->id;
    }

    function getNome() {
        return $this->nome;
    }

    function getNit() {
        return $this->nit;
    }

    function getCpf() {
        return $this->cpf;
    }

    function getRg() {
        return $this->rg;
    }

    function getRgOrgaoExpeditor() {
        return $this->rgOrgaoExpeditor;
    }

    function getRgUfOrgaoExpeditor() {
        return $this->rgUfOrgaoExpeditor;
    }

    function getRgDataExpedicao() {
        return $this->rgDataExpedicao;
    }

    function getNomeMae() {
        return $this->nomeMae;
    }

    function getNomePai() {
        return $this->nomePai;
    }

    function getDataNascimento() {
        return $this->dataNascimento;
    }

    function getSexo() {
        return $this->sexo;
    }

    function getNaturalidade() {
        return $this->naturalidade;
    }

    function getFormaPagamento() {
        return $this->formaPagamento;
    }

    function getBanco() {
        return $this->banco;
    }

    function getAgencia() {
        return $this->agencia;
    }

    function getOperacao() {
        return $this->operacao;
    }

    function getDigitoAgencia() {
        return $this->digitoAgencia;
    }

    function getConta() {
        return $this->conta;
    }

    function getDigitoConta() {
        return $this->digitoConta;
    }

    function getTipoConta() {
        return $this->tipoConta;
    }

    function getEndereco() {
        return $this->endereco;
    }

    function setId($id) {
        $this->id = $id;
        return $this;
    }

    function setNome($nome) {
        $this->nome = $nome;
        return $this;
    }

    function setNit($nit) {
        $this->nit = $nit;
        return $this;
    }

    function setCpf($cpf) {
        $this->cpf = $cpf;
        return $this;
    }

    function setRg($rg) {
        $this->rg = $rg;
        return $this;
    }

    function setRgOrgaoExpeditor($rgOrgaoExpeditor) {
        $this->rgOrgaoExpeditor = $rgOrgaoExpeditor;
        return $this;
    }

    function setRgUfOrgaoExpeditor($rgUfOrgaoExpeditor) {
        $this->rgUfOrgaoExpeditor = $rgUfOrgaoExpeditor;
        return $this;
    }

    function setRgDataExpedicao($rgDataExpedicao) {
        $this->rgDataExpedicao = $rgDataExpedicao;
        return $this;
    }

    function setNomeMae($nomeMae) {
        $this->nomeMae = $nomeMae;
        return $this;
    }

    function setNomePai($nomePai) {
        $this->nomePai = $nomePai;
        return $this;
    }

    function setDataNascimento($dataNascimento) {
        $this->dataNascimento = $dataNascimento;
        return $this;
    }

    function setSexo($sexo) {
        $this->sexo = $sexo;
        return $this;
    }

    function setNaturalidade($naturalidade) {
        $this->naturalidade = $naturalidade;
        return $this;
    }

    function setFormaPagamento($formaPagamento) {
        $this->formaPagamento = $formaPagamento;
        return $this;
    }

    function setBanco($banco) {
        $this->banco = $banco;
        return $this;
    }

    function setAgencia($agencia) {
        $this->agencia = $agencia;
        return $this;
    }

    function setOperacao($operacao) {
        $this->operacao = $operacao;
        return $this;
    }

    function setDigitoAgencia($digitoAgencia) {
        $this->digitoAgencia = $digitoAgencia;
        return $this;
    }

    function setConta($conta) {
        $this->conta = $conta;
        return $this;
    }

    function setDigitoConta($digitoConta) {
        $this->digitoConta = $digitoConta;
        return $this;
    }

    function setTipoConta($tipoConta) {
        $this->tipoConta = $tipoConta;
        return $this;
    }

    function setEndereco($endereco) {
        $this->endereco = $endereco;
        return $this;
    }

    function getUfNaturalidade() {
        return $this->ufNaturalidade;
    }

    function setUfNaturalidade($ufNaturalidade) {
        $this->ufNaturalidade = $ufNaturalidade;
        return $this;
    }

    function getTelefone() {
        return $this->telefone;
    }

    function setTelefone($telefone) {
        $this->telefone = $telefone;
        return $this;
    }

}

File: backend-java/src/main/java/br/com/oljsoft/sgr/model/Pessoa.java
package br.com.oljsoft.sgr.model;

import java.io.Serializable;
import java.text.ParseException;
import java.util.Date;
import java.util.List;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import br.com.oljsoft.sgr.utils.DateUtils;
import br.com.oljsoft.sgr.utils.enuns.EnumBancos;
import br.com.oljsoft.sgr.utils.enuns.EnumFormaPagamentos;
import br.com.oljsoft.sgr.utils.enuns.EnumSexo;
import br.com.oljsoft.sgr.utils.enuns.EnumTipoContas;
import br.com.oljsoft.sgr.utils.enuns.EnumUf;

@Entity
@Table(name = "tab_pessoa")
@JsonIgnoreProperties({ "recibos", "telefones" })
public class Pessoa implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(name = "pess_id")
	private Long id;

	@Column(name = "pess_nome")
	private String nome;

	@Column(name = "pess_nit")
	private String nit;

	@Column(name = "pess_cpf")
	private String cpf;

	@Column(name = "pess_nomepai")
	private String nomePai;

	@Column(name = "pess_nomemae")
	private String nomeMae;

	@Temporal(TemporalType.DATE)
	@Column(name = "pess_datanascimento")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy", locale = "pt-BR", timezone = "Brazil/East")
	private Date dataNascimento;

	@Enumerated(EnumType.ORDINAL)
	@Column(name = "pess_sexo")
	private EnumSexo sexo;

	@Column(name = "pess_naturalidade")
	private String naturalidade;

	@Enumerated(EnumType.ORDINAL)
	@Column(name = "pess_ufnaturalidade")
	private EnumUf ufNaturalidade;

	@Column(name = "pess_numerocartaosus")
	private String numeroCartaoSus;

	@Enumerated(EnumType.ORDINAL)
	@Column(name = "pess_formapagamento")
	private EnumFormaPagamentos formaPagamento;

	@Enumerated(EnumType.ORDINAL)
	@Column(name = "pess_banco")
	private EnumBancos banco;

	@Column(name = "pess_agencia")
	private String agencia;

	@Column(name = "pess_digitoagencia")
	private String digitoAgencia;

	@Column(name = "pess_conta")
	private String conta;

	@Column(name = "pess_digitoconta")
	private String digitoConta;

	@Enumerated(EnumType.ORDINAL)
	@Column(name = "pess_tipoconta")
	private EnumTipoContas tipoConta;

	@OneToOne(mappedBy = "pessoa", fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true)
	@JsonManagedReference
	private PessoaDocumento documento;

	@OneToMany(mappedBy = "pessoa", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	private Set<PessoaTelefone> telefones;

	@OneToOne(mappedBy = "pessoa", fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true)
	@JsonManagedReference
	private PessoaEndereco endereco;

	@OneToMany(mappedBy = "pessoa")
	private List<Recibo> recibos;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public List<Recibo> getRecibos() {
		return recibos;
	}

	public void setRecibos(List<Recibo> recibos) {
		this.recibos = recibos;
	}

	public EnumTipoContas getTipoConta() {
		return tipoConta;
	}

	public void setTipoConta(EnumTipoContas tipoConta) {
		this.tipoConta = tipoConta;
	}

	public EnumBancos getBanco() {
		return banco;
	}

	public void setBanco(EnumBancos banco) {
		this.banco = banco;
	}

	public String getAgencia() {
		return agencia;
	}

	public void setAgencia(String agencia) {
		this.agencia = agencia;
	}

	public String getDigitoAgencia() {
		return digitoAgencia;
	}

	public void setDigitoAgencia(String digitoAgencia) {
		this.digitoAgencia = digitoAgencia;
	}

	public String getConta() {
		return conta;
	}

	public void setConta(String conta) {
		this.conta = conta;
	}

	public String getDigitoConta() {
		return digitoConta;
	}

	public void setDigitoConta(String digitoConta) {
		this.digitoConta = digitoConta;
	}

	public EnumFormaPagamentos getFormaPagamento() {
		return formaPagamento;
	}

	public void setFormaPagamento(EnumFormaPagamentos formaPagamento) {
		this.formaPagamento = formaPagamento;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getNit() {
		return nit;
	}

	public void setNit(String nit) {
		this.nit = nit;
	}

	public String getCpf() {
		return cpf;
	}

	public void setCpf(String cpf) {
		this.cpf = cpf;
	}

	public String getNomePai() {
		return nomePai;
	}

	public void setNomePai(String nomePai) {
		this.nomePai = nomePai;
	}

	public String getNomeMae() {
		return nomeMae;
	}

	public void setNomeMae(String nomeMae) {
		this.nomeMae = nomeMae;
	}

	public Date getDataNascimento() {
		try {
			return DateUtils.parse(this.dataNascimento.toString());
		} catch (ParseException e) {
			e.printStackTrace();
			return null;
		}
	}

	public void setDataNascimento(Date dataNascimento) {
		this.dataNascimento = dataNascimento;
	}

	public EnumSexo getSexo() {
		return sexo;
	}

	public void setSexo(EnumSexo sexo) {
		this.sexo = sexo;
	}

	public String getNaturalidade() {
		return naturalidade;
	}

	public void setNaturalidade(String naturalidade) {
		this.naturalidade = naturalidade;
	}

	public EnumUf getUfNaturalidade() {
		return ufNaturalidade;
	}

	public void setUfNaturalidade(EnumUf ufNaturalidade) {
		this.ufNaturalidade = ufNaturalidade;
	}

	public String getNumeroCartaoSus() {
		return numeroCartaoSus;
	}

	public void setNumeroCartaoSus(String numeroCartaoSus) {
		this.numeroCartaoSus = numeroCartaoSus;
	}

	public PessoaDocumento getDocumento() {
		return documento;
	}

	public void setDocumento(PessoaDocumento documento) {
		this.documento = documento;
	}

	public Set<PessoaTelefone> getTelefones() {
		return telefones;
	}

	public void setTelefones(Set<PessoaTelefone> telefones) {
		this.telefones = telefones;
	}

	public PessoaEndereco getEndereco() {
		return endereco;
	}

	public void setEndereco(PessoaEndereco endereco) {
		this.endereco = endereco;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Pessoa other = (Pessoa) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}